### PR TITLE
Story/16/advanced search

### PR DESCRIPTION
**Pull request by danielnmai**
Advance search filter function is completed. The user list can be filtered based on location, training period, experience and gender.

### CODE REVIEW

File: src/common/Navbar.js [hunk 1]
```diff
@@ -1,33 +1,102 @@
...
   onItemClicked(item) {
+    const { fetchUserInfo, fetchUserFeeds } = this.props;
+
     const sender_id = sessionStorage.getItem('userId');
     const receiver_id = item.value
-    this.props.fetchUserInfo(item.value);
-    this.props.fetchUserFeeds(item.value);
+    fetchUserInfo(item.value);
+    fetchUserFeeds(item.value);
     this.props.fetchUserConnections(item.value);
```
Comment (hypatiah) [inline src/common/Navbar.js]: deconstruct lines 55-57

File: src/reducers/all_users_reducer.js [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
+      return {
+        ...state,
+        filterUserList: action.payload.data,
+        userList: action.payload.data
+      };
+    case FILTER_USERS:
+      var item = action.payload.item;
+      var nestedItem = action.payload.nestedItem;
+      var filterUserList = state.userList;
+      if(item) {
```
Comment (hypatiah) [inline src/reducers/all_users_reducer.js]: nit: add space after `if`

Comment (danielnmai) [reply src/reducers/all_users_reducer.js]: Fixed with latest commit.

File: src/common/Navbar.js (hunk 1 above)
Comment (danielnmai) [reply src/common/Navbar.js]: Fixed with latest commit.

File: src/LoginPage/components/RegisterForm.js [hunk 3]
```diff
@@ -110,7 +110,7 @@ class RegisterForm extends Component {
   }
 
   onSubmit(values) {
-    axios.post('https://qa-womenincomedy.herokuapp.com/api/v1/users', values)
+    axios.post('http://localhost:9000/api/v1/users', values)
```
[the remark this answers is not in the document]
Comment (danielnmai) [reply src/LoginPage/components/RegisterForm.js]: Fixed with latest commit.

**Pull request merged**